### PR TITLE
pkg/otlp: Add `instrumentation_scope_metadata_as_tags`

### PR DESCRIPTION
**Pull request by keisku**
### What does this PR do?

Add a new flag named `instrumentation_scope_metadata_as_tags` to deprecate the current flag, [instrumentation_library_metadata_as_tags](https://github.com/DataDog/datadog-agent/blob/3cd1424c0e2a910187fff3a402b8472c86932503/pkg/config/config_template.yaml#L3197-L3201).

### Motivation

Since https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v0.15.0, InstrumentationLibrary had been deprecated and https://github.com/open-telemetry/opentelemetry-collector/pull/5219 removed it eventually.

This is confusing since the rest of the project has moved to the new name, so we should introduce a new flag that uses the instrumentation scope name, and deprecate the current one.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/otlp/internal/serializerexporter/exporter.go
Comment (mx-psi) [inline pkg/otlp/internal/serializerexporter/exporter.go:106]: I think if the user sets both `InstrumentationScopeMetadataAsTags` and `InstrumentationLibraryMetadataAsTags` we should error out and tell the user to only use the instrumentation scope one

Comment (mx-psi) [reply pkg/otlp/internal/serializerexporter/exporter.go:106]: It's also fine to keep the current behavior, but then we should update the release note: the two options work independently, don't they?

Comment (keisku) [reply pkg/otlp/internal/serializerexporter/exporter.go:106]: Do you mean it should return error like this?

```go
	if cfg.Metrics.ExporterConfig.InstrumentationLibraryMetadataAsTags && cfg.Metrics.ExporterConfig.InstrumentationScopeMetadataAsTags {
		return nil, fmt.Errorf("cannot use both instrumentation_library_metadata_as_tags(deprecated) and instrumentation_scope_metadata_as_tags")
	}
```

Comment (mx-psi) [reply pkg/otlp/internal/serializerexporter/exporter.go:106]: yeah that makes sense!

**Review by mx-psi — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]
LGTM for agent-core owned files (config template)

File: pkg/otlp/internal/serializerexporter/config.go
Comment (cswatt) [inline pkg/otlp/internal/serializerexporter/config.go:151]:
```suggestion
	// Both must not be set at the same time.
```

**Review by cswatt — APPROVED** [review]
leaving a tiny edit, approving so as not to block on it

**Pull request merged**
